### DISCUSSION TITLE
20220515

### DISCUSSION THREAD

**Discussion by tmiw** [ARSFI/paclink-open-source#22] [General]
This is a pre-production build intended to gauge current status of the project and determine what needs to be fixed. Brief testing indicates that packet (using Kenwood TM-D700A), AGW (using AGWPE and UZ7HO soundmodem) and Telnet channels work; Pactor is TBD.

Contains (vs. [last pre-release version](https://github.com/ARSFI/paclink-open-source/releases/tag/nightly20220415)):

* Pactor: increase maximum values for FSK and PSK to 9000mv.
* Pactor: add support for Long Path connections.

Instructions:

1. Unzip paclink-open-source.zip somewhere on your hard drive.
2. Run setup.exe to install on your system. Paclink will be installed in the Start menu.
3. Open Paclink from the Start menu and reconfigure if required. (There is no import functionality from the existing Paclink at this time but if you used a pre-release version, configuration should be saved.)

Please leave comments in the release's discussion thread or email K6AQ@winlink.org with feedback. 👍 

<hr /><em>This discussion was created from the release <a href='https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220515'>20220515</a>.</em>

**Comment by kd4vrf** [comment]
Continuation of thread 20220415.

I've been able to make 7+ connections to an RMS gateway using build 20220515 over the course of about 8 hours. The two Pactor changes (FSK/PSK levels and Long Path support) work flawless for me. I've been able to connect to an RMS gateway every time and the audio levels coming from the Pactor modem are consistent.

I have noticed two issues and have three different types of crashes during the time I've been running Paclink today.

Two Issues:

- Clicking on "help-->contents" does not display any help windows. If this is a known issue or that code is incomplete ignore this one. I didn't see anything under the existing issues related to help so I wanted to mention it.
- "USB Dial:" in the Pactor connect dialog box displays "-----". I believe it should be displaying the "RF Center Freq" minus the "Audio Tones Center:" value specified in the "Pactor TNC Channels" settings dialog box. In the example below the RF center frequency is "7102.000". The dial frequency is 7100.500 (7102.000 MHz - 1500Hz "Audio Tones Center"). When setting the radio manually the conversion from the displayed center frequency isn't difficult but having the dial frequency listed would make it a little easier. Since I have a Flex radio I'm stuck with manual radio adjustments until you get hamlib into Paclink.

![USB Dial Example Screen Shot 2022-05-16 at 3 20 15 PM](https://user-images.githubusercontent.com/45251519/168696363-3f011275-e9fd-4805-958c-9f52515921de.png)

Three Crashes:

- Unhandled Exception - I saw this twice today. The second time I clicked the details button in the error message and captured the output. I have attached that output to this post. Both times I saw this message was when I clicked the "close" button on the Pactor connect dialog box. I was able to close it many times without this error, so it doesn't occur every time.
![Unhandled Exception Screen Shot 2022-05-16 at 1 17 19 PM](https://user-images.githubusercontent.com/45251519/168698513-b51cfbe2-08d9-4e67-aeb9-b83ea0b3c9b0.png)

- "Paclink has stopped working" - I've seen this error 3 times. It doesn't seem to have any pattern. I have seen it as early as a few minutes into launching Paclink, and sometimes it shows up hours after it's been running without me doing anything. The only option is to close Paclink and restart it again. After the restart it functions normally
![Stopped Working error 2 Screen Shot 2022-05-16 at 2 51 18 PM](https://user-images.githubusercontent.com/45251519/168698475-cb560c6a-ca27-45cd-b41f-402c73681739.png)

- Closing Pactor connect dialog box - I saw this error twice today. It occurs after a successful Pactor connection to an RMS gateway. As you can see in the status window it displays "*** Closing Pactor - K4HSP at 2022/05/16 18:08:11". During a successful close the Pactor modem will send the CWid of the station that initiated the connection and then display the Pactor connect dialog box so you can do another connection. In this case it doesn't send the CWid and it doesn't display the Pactor Connection dialog box. The first time it occurred I just closed Paclink. The second time it occurred I clicked abort, when I tried the connection again as soon as Paclink connected to the Pactor modem it sent the CWid. I was able to initiate a second successful connection during that session after I clicked abort.
![Closing Pactor Hung Screen Shot 2022-05-16 at 2 11 56 PM](https://user-images.githubusercontent.com/45251519/168698542-697cb37b-ee49-42ac-9ae6-1d5e737ce785.png)

Here is a .zip archive of the log folders:
[Logs.zip](https://github.com/ARSFI/paclink-open-source/files/8703845/Logs.zip)

Here is the full text from the unhandled exception error:
[Unhandled exception error.txt](https://github.com/ARSFI/paclink-open-source/files/8703851/Unhandled.exception.error.txt)

Let me know if you need anything else or if there is anything else I can test.

**Reply by tmiw** [reply]
Sorry for the delay, I was traveling to and from Dayton and just got time to look into this further.

1. Re: help--Paclink.chm is now included with the installer. I haven't tested this much yet but I imagine it shouldn't require much more to have Help->Show Contents work if it's still not displaying properly.
2. USB Dial frequency: I updated the code to always display this (it was previously only displayed if CAT control was properly configured).
3. Pactor-related crashes: I made some changes that may help with some of the ones you've seen. I can look into more detail once we figure out what's left.

Anyway, the build with the above changes is at https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220525. Thanks again for the testing for sure!

**Reply by kd4vrf** [reply]
No worries about the delay, hopefully you enjoyed Dayton.

I've done some testing today, I'll post my comments on the new release thread.
